llvm/lib/ProfileData/MemProfReader.cpp

```diff
@@ -610,13 +671,30 @@ RawMemProfReader::peekBuildIds(MemoryBuffer *DataBuffer) {
   return BuildIds.takeVector();
 }
 
+llvm::SmallVector<std::pair<uint64_t, MemInfoBlock>>
+RawMemProfReader::readMemInfoBlocks(const char *Ptr) {
```

Comment:
Can you add a FIXME here or somewhere else relevant that we should implement the Schema used by the indexed format to self describe which MIB fields exist in the raw profile, then gracefully handle the reading of older versions?